adapters/repos/db/index_queue.go

```diff
@@ -473,6 +476,8 @@ func (q *IndexQueue) pushToWorkers(max int, wait bool) int64 {
 			}
 		}
 
+		q.queue.ResetDeleted(deleted...)
```

Comment:
I can see this change in the `pushToWorkers` method, this won't break the existing logic, yes?

Reply:
No it should not, this ensures we always clean up the blacklist before dequeuing